crates/gitql-engine/src/engine_function.rs

```diff
@@ -286,19 +291,27 @@ fn select_diffs(
                     .next()
                     .map(|id| id.object().unwrap().into_commit().tree().unwrap())
                     .unwrap_or_else(|| repo.empty_tree());
+
+                let select_insertions_or_deletions =
+                    field_name == "insertions" || field_name == "deletions";
+
```

Comment:
As re-use of the cache across commits is unlikely and since they keep growing, they should be cleared once in a while. Once per commit should do the trick.

Just like [it's done here](https://github.com/Byron/gitoxide/blob/51c7abc65f368b1b2bd3d82473793d3cd4fcbad5/gitoxide-core/src/query/engine/update.rs#L207-L208).

Reply:
Done, Once you approve PR i will merge

Thank you